Sources/NetworkProtection/Networking/NetworkProtectionClient.swift

```diff
@@ -248,12 +256,19 @@ final class NetworkProtectionBackendClient: NetworkProtectionClient {
         do {
             let (data, response) = try await URLSession.shared.data(for: request)
             guard let response = response as? HTTPURLResponse else {
-                return .failure(.failedToFetchRegisteredServers(nil))
+                return .failure(.noResponseFromRegisterEndpoint)
             }
             switch response.statusCode {
-            case 200: responseData = data
-            case 401: return .failure(.invalidAuthToken)
-            default: return (response.statusCode == 403 && isSubscriptionEnabled) ? .failure(.accessDenied) : .failure(.failedToFetchRegisteredServers(nil))
+            case 200:
+                responseData = data
+            case 401:
+                return .failure(.invalidAuthToken)
+            case 403 where isSubscriptionEnabled:
+                return .failure(.accessDenied)
+            default:
+                let domain = #function
+                let error = NSError(domain: domain, code: response.statusCode)
+                return .failure(.unexpectedStatusFromRegisterEndpoint(error))
```

Comment:
Two changes here:

1. Made the code slightly more readable by adding a few newlines and separating a new case.
2. Default case no longer failing with `.failedToFetchRegisteredServers(nil)` as that was used in a few places and doesn't include status code information.  This pixel is not tracked individually so no need for a new triage.